### PR TITLE
Extract frame_system SignedExtensions into separate files.

### PR DESCRIPTION
**Pull request by tomusdrw**
All `SignedExtensions` of `frame_system` used to be placed in one file `frame/system/src/lib.rs` together with the module implementation and the tests.
The file became rather large over time and the code & tests mixed up.

The PR moves each extension to it's own module under `extensions` directory and moves corresponding tests there as well. The test initialisation code is extracted to `mock.rs` file, similar to how other `pallets` are structured.

I didn't change any logic, nor did it change any APIs (extensions are re-exported), it only introduced a bunch of changes necessary to make the code work with the new structure (`pub(crate)`, etc).

Done because I want to change how weights are parametrized and realised that I'd need to add a bunch of new lines to this enormous file, so rather decided to split it first.

Polkadot Companion: https://github.com/paritytech/polkadot/pull/1306

### CODE REVIEW

**Comment by tomusdrw** [comment]
@shawntabrizi tests moved as well.

File: frame/system/src/extensions/check_era.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	traits::{SignedExtension, DispatchInfoOf, SaturatedConversion},
+	transaction_validity::{
+		ValidTransaction, TransactionValidityError, InvalidTransaction, TransactionValidity,
+	},
+
+};
+
+/// Check for transaction mortality.
+#[derive(Encode, Decode, Clone, Eq, PartialEq)]
+pub struct CheckEra<T: Trait + Send + Sync>(Era, sp_std::marker::PhantomData<T>);
```
Comment (bkchr) [inline frame/system/src/extensions/check_era.rs:32]: If we move this, can we may directly rename this? 

And call it `CheckMortality` to make it more obvious what this is doing? I first always think that this checks the era.

File: frame/system/src/extensions/check_era.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+use codec::{Encode, Decode};
+use crate::{Trait, Module, BlockHash};
+use frame_support::StorageMap;
+use sp_runtime::{
+	generic::Era,
+	traits::{SignedExtension, DispatchInfoOf, SaturatedConversion},
+	transaction_validity::{
+		ValidTransaction, TransactionValidityError, InvalidTransaction, TransactionValidity,
+	},
+
```
Comment (bkchr) [inline frame/system/src/extensions/check_era.rs:27]:
```suggestion
```

File: frame/system/src/extensions/check_mortality.rs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		Ok(())
+	}
+}
+
+impl<T: Trait + Send + Sync> SignedExtension for CheckMortality<T> {
+	type AccountId = T::AccountId;
+	type Call = T::Call;
+	type AdditionalSigned = T::Hash;
+	type Pre = ();
+	const IDENTIFIER: &'static str = "CheckEra";
```
Comment (bkchr) [inline frame/system/src/extensions/check_mortality.rs:57]:
```suggestion
	const IDENTIFIER: &'static str = "CheckMortality";
```

CC @jacogr

**Comment by jacogr** [comment]
This is fine - but let's not rush this through to the networks since it needs an API and extension rollout (with approvals). The signed extensions are used in there to make sense of the payload.

**Comment by tomusdrw** [comment]
CC @jacogr the `IDENTIFIER` stays the same now, are you okay with the PR then?

**Pull request merged**